ckanext/opendata_theme/processor.py

```diff
@@ -0,0 +1,397 @@
+from abc import abstractmethod
+from collections import defaultdict, OrderedDict
+import wcag_contrast_ratio as contrast
+
+__all__ = ['custom_style_processor']
+
+from ckanext.opendata_theme.color_conrast import get_contrast
+
+
+class AbstractParser:
+    color = None
+
+    @abstractmethod
+    def get_css_from_data(self, data):
+        return
+
+    def parse_form_data(self, data):
+        value = data.get(self.get_form_name())
+        self.color = value
+
+    @abstractmethod
+    def get_class_name(self):
+        raise NotImplemented
+
+    @abstractmethod
+    def get_form_name(self):
+        raise NotImplemented
+
+    @abstractmethod
+    def get_title(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_default_color(self):
+        raise NotImplementedError
+
+
+class AccountHeaderBackGroundColor(AbstractParser):
+    def get_css_from_data(self, data):
```

Comment:
this method is almost the same for all the classes. we can extract one field from this and change it in all the classes.